src/libfabric/fabric_error.hpp

```diff
@@ -0,0 +1,66 @@
+/*
+ * GridTools
+ *
+ * Copyright (c) 2014-2020, ETH Zurich
+ * All rights reserved.
+ *
+ * Please, refer to the LICENSE file in the root directory.
+ * SPDX-License-Identifier: BSD-3-Clause
+ *
+ */
+#pragma once
+
+#include <stdexcept>
+#include <string>
+#include <string.h>
+//
+#include <rdma/fi_errno.h>
+//
+#include "./print.hpp"
+
+namespace oomph {
+    // cppcheck-suppress ConfigurationNotChecked
+    static hpx::debug::enable_print<true> err_deb("ERROR__");
+}
+
+namespace oomph { namespace libfabric
+{
+
+    class fabric_error : public std::runtime_error
+    {
+    public:
+        // --------------------------------------------------------------------
+        fabric_error(int err, const std::string &msg)
+            : std::runtime_error(std::string(fi_strerror(-err)) + msg),
+              error_(err)
+        {
+            err_deb.error(msg, ":", fi_strerror(-err));
+            std::terminate();
```

Comment:
We do not currently have a mechanism for recovering from hard errors which transition the libfabric endpoint into an error state, so it is really just to force the dump of an error and immediately terminate, so that fixing whatever caused the error is #no. 1 priority. Termination is not necessary, but after the error, the program will probably fail anyway ...